src/permission/PermissionParser.php

```diff
@@ -31,6 +31,10 @@ class PermissionParser{
 
 	public const DEFAULT_OP = "op";
 	public const DEFAULT_NOT_OP = "notop";
+
+	public const DEFAULT_CONSOLE = "console";
+
+	public const DEFAULT_NOT_CONSOLE = "notconsole";
```

Comment:
Don't bother with this. You haven't implemented support for it and no one will use it. (Trust me, no one understands what `notop` is supposed to be used for either.)

Reply:
The "notconsole" may be useful for commands that are not supported to run in cli mode. For example, a command that opens a form for the sender

Reply:
Inverted permissions aren't well-defined.

I also don't think denying permission is the right way to address that, since it'll give confusing messages.

Reply:
You'd also still end up needing to check instanceof of command sender within the execute() anyway to ensure that you're not letting in other types of command senders, or console senders which have had this permission overridden to true.